lib/permutations.rb

```diff
@@ -1,4 +1,31 @@
-
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented" 
+  array_1 = string1.split(//)
+  array_2 = string2.split(//)
+  hash = {}
+
+  if array_1.length == 0 && array_2.length == 0
+    return true
+  elsif array_1.length != array_2.length
+    return false
+  end
+
+  array_1.each do |element|
+    if array_1.count(element) != array_2.count(element)
+      return false
+    end
+  end
+
+  index = 0
+  array_2.each do |element|
+    hash[element] = true
+    index += 1
+  end
```

Comment:
What is `index` doing here?  Also you don't just want to set hash[element] to true, the 2nd loop goes throught he same array and checks to see if `hash[letter]` is true or not.  

Maybe consider:

```ruby
  letter_counts = {}
  array_1.each do |letter|
    letter_counts[letter] = letter_counts[letter] ? letter_counts[letter] + 1 : 1
  end

  array_2.each do |letter|
    if letter_counts[letter] && letter_counts[letter] > 0
      letter[counts] -= 1
    else
      return false
    end
  end

return true if letter_counts.values.all? {|value| value == 0 }